src/components/DAG/DAGUtils.ts

```diff
@@ -18,6 +18,8 @@ export type DAGModelItem = {
 export type StepInfoModel = Record<string, DAGModelItem>;
 export type GraphStructureModel = string | Array<GraphStructureModel>;
 
+export type Decorator = { name: string; attributes: Record<string, string>; statically_defined: boolean };
```

Comment:
@savingoyal Does attributes here have any more defined type definition or can it be anything depending on user?

Reply:
It could be any arbitrary map.

Reply:
We shouldn't hardcode the timeout to 10 mins but instead rely on the timeout specified in attributes here + 30 seconds.